sdk/ml/azure-ai-ml/azure/ai/ml/entities/_assets/_artifacts/_package/model_package.py

```diff
@@ -0,0 +1,292 @@
+# ---------------------------------------------------------
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# ---------------------------------------------------------
+
+# pylint: disable=protected-access, redefined-builtin
+
+from os import PathLike
+from pathlib import Path
+from typing import IO, AnyStr, List, Dict, Optional, Union
+
+
+from azure.ai.ml._restclient.v2023_02_01_preview.models import (
+    PackageRequest,
+    PackageResponse,
+    InferencingServer,
+    ModelPackageInput as RestModelPackageInput,
+    PackageInputPathId as RestPackageInputPathId,
+    PackageInputPathVersion as RestPackageInputPathVersion,
+    PackageInputPathUrl as RestPackageInputPathUrl,
+    BaseEnvironmentSource,
+    ModelConfiguration,
+    CodeConfiguration,
+)
+from azure.ai.ml.entities._resource import Resource
+from azure.ai.ml._schema.assets.package.model_package import ModelPackageSchema
+from azure.ai.ml.entities._util import load_from_dict
+from azure.ai.ml.constants._common import (
+    BASE_PATH_CONTEXT_KEY,
+    PARAMS_OVERRIDE_KEY,
+)
+from azure.ai.ml._utils.utils import snake_to_pascal, dump_yaml_to_file
+from azure.ai.ml._utils._experimental import experimental
+
+
+@experimental
+class PackageInputPathId:
+    def __init__(self, input_path_type: Optional[str] = None, resource_id: Optional[str] = None):
+        """PackageInputPathId.
+
+        :param input_path_type: The type of the input path.
```

Comment:
It would be helpful to say what possible values are accepted here in the doc strings.

Reply:
will do